### DISCUSSION TITLE
How can I achieve OrWhere

### DISCUSSION THREAD

**Discussion by tsndp** [spatie/laravel-query-builder#561] [Q&A]
Taking from official Docs:

```
Basic usage
// GET /users?filter[name]=john&filter[email]=gmail

$users = QueryBuilder::for(User::class)
    ->allowedFilters(['name', 'email'])
    ->get();

// $users will contain all users with "john" in their name AND "gmail" in their email address
```

How can i get below data:
$users contain all users with "john" in their name **OR** "gmail" in their email address

**Comment by vnxx** [comment]
Hi, I got the same issue as u, when I look for the answer I found #611 
So basically u can just make a [Custom filters ](https://spatie.be/docs/laravel-query-builder/v3/features/filtering#custom-filters)

The default for allowedFilters:
[https://github.com/spatie/laravel-query-builder/blob/master/src/Filters/FiltersPartial.php#L28](https://github.com/spatie/laravel-query-builder/blob/master/src/Filters/FiltersPartial.php#L28)

change the `where` to `orWhere`

this works for me:
`$query->orWhere($property, 'LIKE', '%' . $value . '%');`

```
use Spatie\QueryBuilder\Filters\Filter;
use Illuminate\Database\Eloquent\Builder;

class FiltersOrWhere implements Filter
{
    public function __invoke(Builder $query, $value, string $property)
    {
        $query->orWhere($property, 'LIKE', '%' . $value . '%');
    }
}
```
